ibis/expr/window.py

```diff
@@ -83,16 +83,34 @@ def _validate_frame(self):
             )
         elif preceding_tuple:
             start, end = self.preceding
+            if end is None:
+                raise com.IbisInputError("preceding end point cannot be None")
             if start is None:
```

Comment:
This check should happen even when start is not None.
```suggestion
             if end < 0:
                 raise com.IbisInputError (
                     "preceding end point must be non-negative"
                 )
             else if start is not None:
```